src/modules/bindmode.rs

```diff
@@ -1,73 +1,64 @@
 use crate::gtk_helpers::IronbarLabelExt;
-use crate::modules::{Module, ModuleInfo, ModuleParts, ModuleUpdateEvent, WidgetContext};
-use crate::{await_sync, glib_recv, module_impl, try_send};
-use color_eyre::{Report, Result};
+use crate::modules::{Module, ModuleInfo, ModuleParts, WidgetContext};
+use crate::{glib_recv, module_impl, module_update, send_async, spawn};
+use color_eyre::Result;
 use gtk::Label;
 use gtk::prelude::*;
 use serde::Deserialize;
-use swayipc_async::ModeEvent;
 use tokio::sync::mpsc;
 use tracing::{info, trace};
 
 #[derive(Debug, Deserialize, Clone)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
-pub struct SwayModeModule {
+pub struct Bindmode {
     // -- Common --
     /// See [truncate options](module-level-options#truncate-mode).
     ///
     /// **Default**: `null`
     pub truncate: Option<TruncateMode>,
 
-    /// See [layout options](module-level-options#layout)
-    #[serde(default, flatten)]
-    layout: LayoutConfig,
-
     /// See [common options](module-level-options#common-options).
     #[serde(flatten)]
     pub common: Option<CommonConfig>,
 }
 
-impl Module<Label> for SwayModeModule {
-    type SendMessage = ModeEvent;
+impl Module<Label> for Bindmode {
+    type SendMessage = BindModeUpdate;
     type ReceiveMessage = ();
 
-    module_impl!("sway_mode");
+    module_impl!("bindmode");
 
     fn spawn_controller(
         &self,
         _info: &ModuleInfo,
         context: &WidgetContext<Self::SendMessage, Self::ReceiveMessage>,
         _rx: mpsc::Receiver<Self::ReceiveMessage>,
     ) -> Result<()> {
-        info!("Sway Mode module started");
-        let tx = context.tx.clone();
+        info!("Bindmode module started");
+
+        let client = context.try_client::<dyn crate::clients::compositor::BindModeClient>()?;
 
-        await_sync(async move {
-            let client = context.ironbar.clients.borrow_mut().sway()?;
-            client
-                .add_listener::<swayipc_async::ModeEvent>(move |mode| {
-                    trace!("mode: {:?}", mode);
-                    try_send!(tx, ModuleUpdateEvent::Update(mode.clone()));
-                })
-                .await?;
+        let tx = context.tx.clone();
 
-            Ok::<(), Report>(())
-        })?;
+        let mut rx = client.subscribe()?;
+        spawn(async move {
+            while let Ok(ev) = rx.recv().await {
+                module_update!(tx, ev);
+            }
+        });
 
         Ok(())
     }
 
     fn into_widget(
         self,
         context: WidgetContext<Self::SendMessage, Self::ReceiveMessage>,
-        info: &ModuleInfo,
+        _info: &ModuleInfo,
     ) -> Result<ModuleParts<Label>> {
-        let label = Label::builder()
-            .use_markup(true)
-            .angle(self.layout.angle(info))
-            .justify(self.layout.justify.into())
-            .build();
+        let label = Label::new(None);
+        label.set_use_markup(true);
```

Comment:
This needs reverting to its original impl:

```suggestion
        let label = Label::builder()
            .use_markup(true)
            .angle(self.layout.angle(info))
            .justify(self.layout.justify.into())
            .build();
```